// usrdb/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/aid.db');

// === usr ===
exports.mailUsr=function(arg2){
var stm=db.prepare('select * from usr where email=?');
var row=stm.get(arg2);
return row;
};
exports.psslUsr=function(arg2){
var stm=db.prepare('select * from usr where pss=?');
var row=stm.get(arg2);
return row;
};
exports.namUsr=function(arg2){
var stm=db.prepare('select * from usr where name=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};
exports.allUsr=function(){
var stm=db.prepare('select * from usr ');
var row=stm.all();
return row;
};
exports.phnUsr=function(arg2){
var stm=db.prepare('select * from usr where phn=?');
try{var row=stm.get(arg2);
}catch(err){console.log(err);}
return row;
};

// === adr ===
exports.mailAdr=function(email){
var stm=db.prepare('select * from adr where email=? ');
try{var row=stm.get(email);}catch(err){console.log(err)}
return row;
};
exports.insAdr=function(email,phn,ln1,ln2,city,sta,zip){
var stm=db.prepare('insert into adr values(?,?,?,?,?,?,?)')
try{stm.run(email,phn,ln1,ln2,city,sta,zip);}catch(err){console.log(err);}
};
exports.phnAdr=function(phn,email,){
var stm=db.prepare('update adr set phn=? where email=?')
try{stm.run(phn,email);}catch(err){console.log(err);}
};
// === usr run ===
exports.updusr=function(uni,usr,sku){
var stm=db.prepare('update usr set uni=? where usr=? and sku=? ');
try{stm.run(uni,usr,sku);
}catch(err){console.log(err);}
};

exports.insUsr=function(name,pss,email){
var stm=db.prepare('insert into usr values(?,?,?) ');
stm.run(name,pss,email);
};

exports.delusr=function(){
var stm=db.prepare('delete from usr where uni=0');
try{stm.run();
}catch(err){var err=err;console.log(err);}
};
// === mer ===
exports.allMer=function(){
var stm=db.prepare('select *,rowid from mer');
try{var row=stm.all();
}catch(err){console.log(err);}
return row;
};

exports.skuMer=function(sku){
var stm=db.prepare('select * from mer where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};

exports.insMer=function(name,sku,pri,img,rel,cat,des){
var stm=db.prepare('insert into mer values(?,?,?,?,?,?,?)');
stm.run(name,sku,pri,img,rel,cat,des);
};

// === pid ===

exports.insPid=function(email,pid,mnt,buy,ite){
var stm=db.prepare('insert into pid values(?,?,?,?,?)');
stm.run(email,pid,mnt,buy,ite)
};

exports.insBuy=function(nam1,email,phn,pid){
var stm=db.prepare('insert into buy values(?,?)');
stm.run(email,pid)
};

exports.upBuy=function(buy,email){
var stm=db.prepare('update pid set buy=? where email=?')
try{stm.run(buy,email)}
catch(err){console.log(err)}
};

exports.selPid=function(email){
var stm=db.prepare('select *,rowid from pid where email=?');
try{var row=stm.all(email);
}catch(err){console.log(err);}
return row;
};
exports.insOrd=function(tax,ship,items){
var stm=db.prepare('insert into ord values(?,?,?)');
stm.run(tax,ship,items)
};



// === aid ===
exports.allAid=function(){
var stm=db.prepare('select * from aid');
try{var row=stm.all();}catch(err){console.log(err);}
return row;
};
exports.insAid=function(usr,pss,email,phn){
var stm=db.prepare('insert into aid values(?,?,?,?)');
stm.run(usr,pss,email,phn);
};
exports.updAid=function(pid,usr){
var stm=db.prepare('update aid set pid=? where usr=?');
stm.run(pid,usr);
};
exports.ordAid=function(usr,pid,name,mnt,dat,stat){
var stm=db.prepare('insert into aid_ord values(?,?,?,?,?,?)');
stm.run(usr,pid,name,mnt,dat,stat);
};
