// cardb/index.js
var Database = require('better-sqlite3');
var db = new Database(__dirname+'/db/car.db');
// === tmp ===
exports.mailTmp=function(arg2){
var stm=db.prepare('select * from tmp where email=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};

exports.allTmp=function(){
var stm=db.prepare('select * from tmp ');
var row=stm.all();
return row;
};

exports.skuTmp=function(sku){
var stm=db.prepare('select * from tmp where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};

exports.usrTmp=function(usr){
var stm=db.prepare('select * from tmp where usr=?');
var row=stm.get(usr);
return row;
};

// === tmp run ===
exports.updTmp=function(uni,email,sku){
var stm=db.prepare('update tmp set uni=? where email=? and sku=? ');
try{stm.run(uni,email,sku);
}catch(err){console.log(err);}
};

exports.insTmp=function(email,sku,uni){
var stm=db.prepare('insert into tmp values(?,?,?) ');
try{stm.run(email,sku,uni);
}catch(err){console.log(err);}
};

exports.delUni=function(){
var stm=db.prepare('delete from tmp where uni=0');
try{stm.run();
}catch(err){console.log(err);}
};

exports.delEma=function(email){
var stm=db.prepare('delete from tmp where email=?');
try{stm.run(email);
}catch(err){console.log(err);}

};

// === son ===

exports.insSon=function(email,son){
var stm=db.prepare('insert into son values(?,?) ');
try{stm.run(email,son);
}catch(err){console.log(err);}
};

exports.mailSon=function(arg2){
var stm=db.prepare('select * from son where email=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};


// === jol ===
exports.getCnt=function(){
var stm=db.prepare('select * from cnt');
var row=stm.get();
return row;
};

exports.updCnt=function(cnt){
var stm=db.prepare('update cnt set cnt=?');
stm.run(cnt);
};

exports.emaJol=function(arg2){
var stm=db.prepare('select * from jol where email=?');
var row=stm.get(arg2);
return row;
};

exports.insJol=function(email,sku,uni,cnt){
var stm=db.prepare('insert into jol values(?,?,?,?) ');
stm.run(email,sku,uni,cnt);
};

exports.delJol=function(){
var stm=db.prepare('delete from jol');
stm.run();
};
// === mer ===
exports.allMer=function(){
var stm=db.prepare('select *,rowid from mer order by sku desc');
try{var row=stm.all();
}catch(err){console.log(err);}
return row;
};

exports.skuMer=function(sku){
var stm=db.prepare('select * from mer where sku=?');
try{var row=stm.get(sku);}
catch(err){console.log(err);}
return row;
};

exports.insMer=function(name,sku,pri,img,rel,cat,des,song){
var stm=db.prepare('insert into mer values(?,?,?,?,?,?,?,?)');
stm.run(name,sku,pri,img,rel,cat,des,song);
};

exports.delMer=function(sku){
var stm=db.prepare('delete from mer where sku=?')
stm.run(sku)
};
exports.namMer=function(name,sku){
var stm=db.prepare('update mer set name=? where sku=?')
try{stm.run(name,sku)}
catch(err){console.log(err)}
};
exports.priMer=function(pri,sku){
var stm=db.prepare('update mer set pri=? where sku=?')
stm.run(pri,sku)
};
exports.relMer=function(rel,sku){
var stm=db.prepare('update mer set rel=? where sku=?')
stm.run(rel,sku)
};
exports.catMer=function(cat,sku){
var stm=db.prepare('update mer set cat=? where sku=?')
stm.run(cat,sku)
};
exports.desMer=function(des,sku){
var stm=db.prepare('update mer set des=? where sku=?')
stm.run(des,sku)
};
exports.sonMer=function(song,sku){
var stm=db.prepare('update mer set song=? where sku=?')
stm.run(song,sku)
};
// === song
exports.updSon=function(jso,sku){
var stm=db.prepare('update mer set song=? where sku=?');
stm.run(jso,sku)

};

exports.skuSon=function(sku){
var stm=db.prepare('select song from mer where sku=?');
try{var row=stm.get(sku);}catch(err){console.log(err);}
return row
};

// === aid ===
exports.allAid=function(){
var stm=db.prepare('select * from aid');
try{var row=stm.all();}catch(err){console.log(err);}
return row;
};
exports.insAid=function(usr,pss,email,phn){
var stm=db.prepare('insert into aid values(?,?,?,?)');
stm.run(usr,pss,email,phn);
};
exports.updAid=function(pid,usr){
var stm=db.prepare('update aid set pid=? where usr=?');
stm.run(pid,usr);
};
exports.ordAid=function(usr,pid,name,mnt,dat,stat){
var stm=db.prepare('insert into aid_ord values(?,?,?,?,?,?)');
stm.run(usr,pid,name,mnt,dat,stat);
};
