// cardb/index.js
var Database = require('better-sqlite3');
//var db = new Database('public/dat/db/car.db');
var db = new Database(__dirname+'/db/car.db');
// === tmp ===
exports.mailTmp=function(arg2){
var stm=db.prepare('select * from tmp where email=?');
try{var row=stm.all(arg2);
}catch(err){console.log(err);}
return row;
};

exports.allTmp=function(){
var stm=db.prepare('select * from tmp ');
var row=stm.all();
return row;
};

exports.skuTmp=function(sku){
var stm=db.prepare('select * from tmp where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};


// === tmp run ===
exports.updTmp=function(uni,email,sku){
var stm=db.prepare('update tmp set uni=? where email=? and sku=? ');
try{stm.run(uni,email,sku);
}catch(err){console.log(err);}
};

exports.insTmp=function(email,sku,uni){
var stm=db.prepare('insert into tmp values(?,?,?) ');
try{stm.run(email,sku,uni);
}catch(err){console.log(err);}
};

exports.delTmp=function(){
var stm=db.prepare('delete from tmp where uni=0');
try{stm.run();
}catch(err){console.log(err);}
};

// === jol ===
exports.getCnt=function(){
var stm=db.prepare('select * from cnt');
var row=stm.get();
return row;
};

exports.updCnt=function(cnt){
var stm=db.prepare('update cnt set cnt=?');
stm.run(cnt);
};

exports.emaJol=function(arg2){
var stm=db.prepare('select * from jol where email=?');
var row=stm.get(arg2);
return row;
};

exports.insJol=function(email,sku,uni,cnt){
var stm=db.prepare('insert into jol values(?,?,?,?) ');
stm.run(email,sku,uni,cnt);
};

exports.delJol=function(){
var stm=db.prepare('delete from jol');
stm.run();
};
// === mer ===
exports.allMer=function(){
var stm=db.prepare('select *,rowid from mer order by sku desc');
try{var row=stm.all();
}catch(err){console.log(err);}
return row;
};

exports.skuMer=function(sku){
var stm=db.prepare('select * from mer where sku=?');
try{var row=stm.get(sku);
}catch(err){console.log(err);}
return row;
};

exports.insMer=function(name,sku,pri,img,rel,cat,des){
var stm=db.prepare('insert into mer values(?,?,?,?,?,?,?)');
stm.run(name,sku,pri,img,rel,cat,des);
};

// === song
exports.altSon=function(song){
var stm=db.prepare('alter table song add colum song');
stm.run(song);
};

// === aid ===
exports.allAid=function(){
var stm=db.prepare('select * from aid');
try{var row=stm.all();}catch(err){console.log(err);}
return row;
};
exports.insAid=function(usr,pss,email,phn){
var stm=db.prepare('insert into aid values(?,?,?,?)');
stm.run(usr,pss,email,phn);
};
exports.updAid=function(pid,usr){
var stm=db.prepare('update aid set pid=? where usr=?');
stm.run(pid,usr);
};
exports.ordAid=function(usr,pid,name,mnt,dat,stat){
var stm=db.prepare('insert into aid_ord values(?,?,?,?,?,?)');
stm.run(usr,pid,name,mnt,dat,stat);
};
